dotnet/src/SemanticKernel.Abstractions/Events/FunctionInvokingEventArgs.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using Microsoft.SemanticKernel.Orchestration;
+using Microsoft.SemanticKernel.SkillDefinition;
+
+namespace Microsoft.SemanticKernel.Events;
+
+/// <summary>
+/// Event arguments available to the Kernel.FunctionInvoking event.
+/// </summary>
+public class FunctionInvokingEventArgs : SKCancelEventArgs
+{
+    /// <summary>
+    /// Indicates if the function execution should be skipped.
+    /// </summary>
+    public bool IsSkipRequested => this._skipRequested;
```

Comment:
Can the rendered prompt be part of the event args?

This would:
- Remove need for `InternalVariables` on `SKContext`
- Enable callers to modify the rendered prompt before invocation (not clear how that is possible now if at all)

Reply:
This was the previous approach and @shawncal did a good comment on it, saying that we don't have to add dedicated `Prompt`

>  As far as I can see, the words "prompt" or "completion" or anything like that... they don't need to be anywhere in the Kernel.Core/Abstractions, because the Kernel shouldn't care what that string is going to be used for. It just says "Run next function"

Reply:
> Enable callers to modify the rendered prompt before invocation (not clear how that is possible now if at all)

Now that's not possible but can be easily achieved adding a `Set` to the `SemanticKernelEventArgs.RenderedPrompt` property

Reply:
Isn't that one of the goals of this PR?
> - Set: Modify a prompt content before sending it to LLM

Reply:
It's part of the Phase 2 implementation.